packages/components/src/templates/next/components/internal/Navbar/NavItem.tsx

```diff
@@ -99,18 +100,42 @@ export const NavItem = forwardRef<HTMLButtonElement, NavbarItemProps>(
 const Megamenu = ({
   name,
   description,
+  referenceLinkHref,
   onCloseMegamenu,
   items,
   LinkComponent,
-}: {
-  name: string
-  description?: string
-  items: BaseNavbarItemProps[]
+}: Pick<
+  BaseNavbarItemProps,
+  "name" | "description" | "referenceLinkHref" | "items"
+> & {
   LinkComponent: NavbarProps["LinkComponent"]
   onCloseMegamenu: () => void
 }) => {
   useScrollLock()
 
+  const renderTitleContent = () => {
+    if (!referenceLinkHref) {
+      return name
+    }
+
+    const isExternal = isExternalUrl(referenceLinkHref)
+    return (
+      <Link
+        LinkComponent={LinkComponent}
+        isExternal={isExternal}
+        showExternalIcon={isExternal}
+        isWithFocusVisibleHighlight
+        href={referenceLinkHref}
+        className="group inline-flex w-fit items-center gap-1.5 hover:text-brand-interaction-hover hover:no-underline"
+      >
+        {name}
+        {!isExternal && (
+          <BiRightArrowAlt className="text-[1.5rem] transition ease-in group-hover:translate-x-2.5" />
```

Comment:
[no action required] i think we shuold've extracted this out as a component hahahahhaha